### PR TITLE
Added decay rate hyperparameter to greedy policy.

### PR DESCRIPTION
**Pull request by abhinavsagar**
Added decay rate in `greedy_policy.hpp` file.

### CODE REVIEW

File: src/mlpack/methods/reinforcement_learning/policy/greedy_policy.hpp [hunk 1]
```diff
@@ -41,13 +42,16 @@ class GreedyPolicy
...
    *        will anneal.
    * @param minEpsilon Epsilon will never be less than this value.
+   * @param decayRate How much to change the model in response to the
+   *        estimated error each time the model weights are updated.
    */
   GreedyPolicy(const double initialEpsilon,
                const size_t annealInterval,
-               const double minEpsilon) :
+               const double minEpsilon,
+               const double decayRate) :
```
Comment (zoq) [inline src/mlpack/methods/reinforcement_learning/policy/greedy_policy.hpp]: To provide backward compatibility we should set a default value of one, that should also fix the build issue.

File: src/mlpack/tests/q_learning_test.cpp [hunk 2]
```diff
@@ -49,7 +49,7 @@ BOOST_AUTO_TEST_CASE(CartPoleWithDQN)
   model.Add<Linear<>>(128, 2);
 
   // Set up the policy and replay method.
-  GreedyPolicy<CartPole> policy(1.0, 1000, 0.1);
+  GreedyPolicy<CartPole> policy(1.0, 1000, 0.1, 0.99);
```
Comment (zoq) [inline src/mlpack/tests/q_learning_test.cpp]: Haven't tested it but does this has any visible effect on the test results?

Comment (abhinavsagar) [reply src/mlpack/tests/q_learning_test.cpp]: @zoq Yes by adding this the change in parameters as they approach the final result is reducing as the iteration proceeds.

**Comment by zoq** [comment]
Thanks for another great contribution.

**Comment by abhinavsagar** [comment]
Thanks.

**Pull request merged**